### PR TITLE
fix RelationalExpression

### PR DESCRIPTION
**Pull request by czpmango**
This PR fixes some bugs about Value comparison related to `null` and `empty`.  Fixes https://github.com/vesoft-inc/nebula-graph/issues/644
There are some test cases:
 - 2>true : null   2<=true : null
 - 2==true : false
 - null==true : null
 - null<>null : null
 - empty==empty : true
 - empty==null : null
 - empty>=1 : empty

### CODE REVIEW

**Review by yixinglu — COMMENTED** [review]
It's not a good practice that fixing bug and refactoring in one PR since i don't clearly check where you have changed.

**Comment by czpmango** [comment]
> It's not a good practice that fixing bug and refactoring in one PR since i don't clearly check where you have changed.

No changes have been made to the behavior of the class `Value`. Just fix RelationalExpression’s behavior, and this requires adding interfaces `Value::lessThan` to allow errors to be passed to the Expression layer.

**Review by yixinglu — APPROVED** [review]

**Review by nevermore3 — DISMISSED** [review]

File: src/common/datatypes/Value.cpp [hunk 1]
```diff
@@ -2226,164 +2402,22 @@ Value operator!(const Value& rhs) {
...
-            return false;
-        }
-        case Value::Type::NULLVALUE:
-        case Value::Type::__EMPTY__: {
-            return false;
-        }
+    auto res = lhs.lessThan(rhs);
+    if (res.isBool()) {
+        return res.getBool();
+    } else {
```
[the remark this answers is not in the document]
Comment (czpmango) [reply src/common/datatypes/Value.cpp]: Already removed.

**Review by yixinglu — DISMISSED** [review]

**Review by nevermore3 — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Review by jievince — APPROVED** [review]

File: src/common/expression/RelationalExpression.cpp [hunk 2]
```diff
@@ -15,31 +15,24 @@ const Value& RelationalExpression::eval(ExpressionContext& ctx) {
...
-            result_ = lhs < rhs;
+            result_ = lhs.lessThan(rhs);
             break;
         case Kind::kRelLE:
-            result_ = lhs <= rhs;
+            result_ = lhs.lessThan(rhs) || lhs.equal(rhs);
             break;
         case Kind::kRelGT:
-            result_ = lhs > rhs;
+            result_ = !lhs.lessThan(rhs) && !lhs.equal(rhs);
```
Comment (Shylock-Hg) [inline src/common/expression/RelationalExpression.cpp:32]: What's the result of `!NULL`?

File: src/common/expression/RelationalExpression.cpp [hunk 3]
```diff
@@ -15,31 +15,24 @@ const Value& RelationalExpression::eval(ExpressionContext& ctx) {
...
-            result_ = lhs != rhs;
+            result_ = !lhs.equal(rhs);
             break;
         case Kind::kRelLT:
-            result_ = lhs < rhs;
+            result_ = lhs.lessThan(rhs);
             break;
         case Kind::kRelLE:
-            result_ = lhs <= rhs;
+            result_ = lhs.lessThan(rhs) || lhs.equal(rhs);
```
Comment (Shylock-Hg) [inline src/common/expression/RelationalExpression.cpp:29]: And why not call <= directly.

**Review by nevermore3 — APPROVED** [review]

File: src/common/expression/RelationalExpression.cpp (hunk 3 above)
Comment (czpmango) [reply src/common/expression/RelationalExpression.cpp:29]: add static method?

**Pull request merged**